tornado-api/src/main/java/uk/ac/manchester/tornado/api/profiler/ChromeEventJSonWriter.java

```diff
@@ -0,0 +1,80 @@
+package uk.ac.manchester.tornado.api.profiler;
+
+import sun.management.VMManagement;
+
+import java.io.File;
+import java.lang.management.ManagementFactory;
+import java.lang.management.RuntimeMXBean;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+
+public class ChromeEventJSonWriter extends JSonWriter<ChromeEventJSonWriter> {
+    static private int pid = 0;
+    ContentWriter NO_ARGS = null;
+
+    ChromeEventJSonWriter() {
+        super();
+        // Java 9 we could use long pid = ProcessHandle.current().pid();
+        // For java 8 we must go through some hoops.
+        try {
+            RuntimeMXBean runtime = ManagementFactory.getRuntimeMXBean();
+            Field jvm = runtime.getClass().getDeclaredField("jvm");
+            jvm.setAccessible(true);
+            VMManagement mgmt = (VMManagement) jvm.get(runtime);
+            java.lang.reflect.Method pid_method = mgmt.getClass().getDeclaredMethod("getProcessId");
+            pid_method.setAccessible(true);
+            pid = (Integer) pid_method.invoke(mgmt);
+        } catch (NoSuchFieldException | IllegalAccessException | NoSuchMethodException | InvocationTargetException nsf) {
+            // well we tried. Lets just use the default 0
+        }
+        objectStart();
+        arrayStart("traceEvents");
+        object(() -> {
```

Comment:
This code answers my first question. So the fact that the API calls this interface automatically sets the JSON file.